server/app/com/xsn/explorer/data/anorm/dao/BlockPostgresDAO.scala

```diff
@@ -316,15 +316,15 @@ class BlockPostgresDAO @Inject()(
         |FROM (
         |   SELECT blockhash, previous_blockhash, next_blockhash, merkle_root, height, time, difficulty
         |   FROM blocks
+        |   ORDER BY height $order
+        |   LIMIT {limit}
         |) blk
         |LEFT JOIN (
         |   SELECT blockhash, count(*) count 
         |   FROM transactions 
         |   GROUP BY blockhash
         |) tx 
         |ON blk.blockhash = tx.blockhash
-        |ORDER BY height $order
```

Comment:
Please explain why this is a problem, and why it gets solved, the ideal why is to submit a test that fails unless you have this code.

Reply:
This was due to the performance problem - instead of querying all the items from `blocks` and do the join and filter, we can do the filter first and do the join. This can improve the performance of the query.